measurePower.py

```diff
@@ -37,29 +38,24 @@ def extract(r: Result) -> int:
             logger.error(f"Could not find Instantaneous power reading: {e}.")
             return 0
 
-        def stat(self, cmd: str, duration: int) -> Result:
-            end_time = time.time() + float(duration)
+        def stat(self, cmd: str) -> Result:
+            SyncManager.wait_on_barrier()
             total_pwr = 0
             iteration = 0
-            while True:
+            while SyncManager.client_not_finished():
                 r = self.run_oc(cmd)
                 if r.returncode != 0:
                     logger.error(f"Failed to get power {cmd}: {r}")
                 pwr = extract(r)
                 total_pwr += pwr
                 iteration += 1
-                # FIXME: Hardcode interval for now
```

Comment:
This is a bit nuanced the reason the FIXME is here is because we want to measure power strictly between the start and stop of the test. E.g.
```
       Start Tests                                     Stop Test
|---------|--------------------------------------------|-------|
     |<---- Measuring power here is not desirable
                                |<------ Measuring power here is desirable
  So what could happen is that you are measuring power here:
                                                          |<---- just a few ms after the client finishes but when you checked before it didn't finish
```
So it might be better to have another sync right before the client is about to end. I know this power measurement is a bit tricky because it measures instantaneous power instead of average power (I believe). If it is average power then it should be ok. E.g.
```
Instantaneous power reading: 162 Watts
Minimum during sampling period: 54 Watts
Maximum during sampling period: 324 Watts
Average power reading over sample period: 132 Watts
```

Reply:
I'm not sure how to sync before the client is about to end.

Maybe we can log the timestamp of when the client finishes, and discard measurements after that? We could do this for MeasureCPU as well

Reply:
Do you think a semaphore will work here? Or we using the inbuilt timeout functionality of iperf?

Reply:
My timing idea is bad. Then we'd have to make sure to synchronize the times between all the pods, and it would be very fragile/sometimes fail for no reason.

I changed measurePower to measure every 0.2 seconds instead of 2 seconds. Because there are so many more measurements, we can probably afford to simply discard the last few power results if they affect the average too much.

I can take a look at how the last few measurements affect the average, if this sounds reasonable.

Reply:
I think this is ok for now.